adapters/medianet/medianet.go

```diff
@@ -0,0 +1,118 @@
+package medianet
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	reqJson, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
```

Comment:
There seems to not be need for `errs` at all if we manage this error scenario just like `MakeBids` does in line 62. We could just:
```
20   func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
21 -     var errs []error
22   
23       reqJson, err := json.Marshal(request)
24       if err != nil {
25 -         errs = append(errs, err)
26 -         return nil, errs
   +         return nil, []error{err}
27       }
28   
adapters/medianet/medianet.go
```

Reply:
Lets keep it in place for direct Bidder request unit test cases, thru that one can test multiple request formats which medianet will accept. If I remove it, then there is not need to test cases and I wont be able to test medianet adapter as an individual module while making tampered requests from unit test cases

Reply:
@guscarreon , https://github.com/prebid/prebid-server/pull/2133#discussion_r792881094 issue addressed, in which I have added `if len(hostUrl) == 0 {
    +         return mnetUrl
    +     }`
    before URL parser